experiments/AtmosLES/squall_line.jl

```diff
@@ -0,0 +1,295 @@
+
+    # Boundary conditions
+    # SGS Filter constants
+    C_smag = FT(0.18) # 0.21 for stable testing, 0.18 in practice
+    C_drag = FT(0.0011)
+    LHF = FT(50)
+    SHF = FT(10)
+    ics = init_squall_line!
+
+    source = (Gravity(), rayleigh_sponge, CreateClouds())
+
+
+    problem = AtmosProblem(
+        boundarycondition = (AtmosBC(), AtmosBC()),
+        init_state_prognostic = ics,
+    )
+
+
+    model = AtmosModel{FT}(
+        AtmosLESConfigType,
+        param_set;
+        problem = problem,
+        ref_state = ref_state,
+        moisture = NonEquilMoist(),
+        turbulence = SmagorinskyLilly{FT}(C_smag),#ConstantViscosityWithDivergence{FT}(200),
+        source = source,
+    )
+
+    ode_solver = ClimateMachine.IMEXSolverType()
+
+    config = ClimateMachine.AtmosLESConfiguration(
+        "Squall_line",
+        N,
+        resolution,
+        xmax,
+        ymax,
+        zmax,
+        param_set,
+        init_squall_line!,
+        xmin = xmin,
+        ymin = ymin,
+        solver_type = ode_solver,
+        model = model,
+        periodicity = (true, true, false),
+        boundary = ((2, 2), (2, 2), (1, 2)),
+        #numerical_flux_first_order = RoeNumericalFlux(),
+    )
+    return config
+end
+function config_diagnostics(driver_config)
+    interval = "10000steps"
+    dgngrp = setup_atmos_default_diagnostics(interval, driver_config.name)
+    return ClimateMachine.DiagnosticsConfiguration([dgngrp])
+end
+
+function main()
+
+    FT = Float64
+
+    # DG polynomial order
+    N = 4
+
+    # Domain resolution and size
+    Δx = FT(250)
+    Δy = FT(1000)
+    Δv = FT(200)
+    resolution = (Δx, Δy, Δv)
+
+    xmax = FT(30000)
+    ymax = FT(5000)
+    zmax = FT(24000)
+    xmin = FT(-30000)
+    ymin = FT(0)
+
+    t0 = FT(0)
+    timeend = FT(9000)
+    spl_tinit, spl_qinit, spl_uinit, spl_vinit, spl_pinit = spline_int()
+    Cmax = FT(0.4)
+    driver_config =
+        config_squall_line(FT, N, resolution, xmax, ymax, zmax, xmin, ymin)
+    solver_config = ClimateMachine.SolverConfiguration(
+        t0,
+        timeend,
+        driver_config,
+        (spl_tinit, spl_qinit, spl_uinit, spl_vinit, spl_pinit);
+        init_on_cpu = true,
+        Courant_number = Cmax,
+    )
+    #dgn_config = config_diagnostics(driver_config)
+
+    cbtmarfilter = GenericCallbacks.EveryXSimulationSteps(1) do (init = false)
+        Filters.apply!(solver_config.Q, (6), solver_config.dg.grid, TMARFilter())
+        nothing
+    end
+    filterorder = 30
+    filter = ExponentialFilter(solver_config.dg.grid, 0, filterorder)
+    cbfilter = GenericCallbacks.EveryXSimulationSteps(1) do
+        Filters.apply!(solver_config.Q, (2, 3, 4), solver_config.dg.grid, filter)
+        nothing
+    end
+    cutoff = CutoffFilter(solver_config.dg.grid)
+    cbcutoff = GenericCallbacks.EveryXSimulationSteps(1) do
+        Filters.apply!(solver_config.Q, (2, 3, 4), solver_config.dg.grid, cutoff)
```

Comment:
Can we grab these tuples of integers with `varsindex`? Otherwise this will be brittle wrt the model changing. It's also opaque as it stands.

Reply:
On top of that, none of these filters are currently being used..

Reply:
I'll remove the filtering since the case doesn't seem to currently need it.

Reply:
Sounds good. It might be that I will need it for precip, but lets see. Maybe not